src/args.rs

```diff
@@ -1,8 +1,14 @@
-#[derive(Debug)]
-pub struct Args {}
+use clap::Parser;
 
-impl Args {
-    pub fn parse() -> Self {
-        todo!()
-    }
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+pub struct Args {
+	/// Application name
+	#[arg(long)]
+	app: String,
```

Comment:
I wonder whether the name of the application will be specified so commonly that it would be better as a mandatory positional argument instead of a mandatory option? This would also nicely clean up the call syntax here :D

Genuine question, is 'app' or 'application' more common in Juju parlance?

Also, if we wanted to generalise this field in to more than just applications, would something like `to_monitor` be more suitably vague?

Reply:
Hmm, deep question :P 
I thought of monitoring metrics of different objects, other than the app, and then perhaps provide options according to what we monitor. Possibilities are cloud, controller, model, unit, relation, etc. "monitor" here doesn't have to be a time-plot of course. I think it is about time we had a look at what metrics are exposed for different juju objects. An option could be to limit the scope of jperf to deployed applications only - maybe the v1 release could be just that :D 

Btw, somehow I lost our work from Tue :'(  so this is from scratch, you might have noticed haha

I guess calling this 'application' is better - changed!

Without specifying a generic "what to monitor" option name, we would have:
`jperf --application APP_NAME`  --> to be followed by options that only make sense for an application
`jperf --cloud CLOUD_NAME` --> to be followed by options that only make sense for a cloud

I have a feeling of a tree of options :D 

Or would we instead stack up the most sensible metrics to monitor as a predefined batch for each type to monitor?

Good idea to have something generic as an option to specify what to monitor. I'm terrible with naming things, so I'll go with `to_monitor`. But this is getting a bit lengthy, with having to say `application` twice:

`jperf --to_monitor application --application APP_NAME`

Just had a quick look at the client facade for metrics, which is slightly oddly named as metricsdebug :P
Looks like we have some metrics collection API for Unit, Application, and Model.

Reply:
I was thinking to keep things simple we could have `jperf <application>` (the `to_monitor` here would just be an internal field name). Having a tree of options sounds dangerous---I'd like to avoid long chains of dependent arguments as the API could get very cluttered! For now though, keeping the scope tight and just handling applications sounds like a good idea.

I hadn't actually noticed, the code looks good!

I'm thinking we should try to show all metrics (or at least as many as possible), but categorising them into TUI tabs so the user doesn't get overwhelmed with info. I think it would be really nice to show the state of the entire system, or at least some large subset of it, for example everything running in a juju deployment, or some particular app, that way, the decision of what the user wants to look at can be deferred to after they've run the command. Hopefully, once a user sees some metrics, they'll get curious and want to investigate more (and not want to wait for enough history to be re-accumulated on graphs if they are forced to stop and re-run `jperf`)

I really dislike having to specify 'application' twice above! If we do go down the route of selecting multiple things, then in the long run, sub-commands would be cleaner. I'll leave the following as something to think about long-term, but for now, let's go with `jperf <application-name>`.
```bash
jperf # Just monitors everything
jperf app my-super-cool-app
jperf cloud localhost-microk8s

jperf show apps
jperf show clouds
```

Reply:
Sounds good!

Reply:
Just for the record, as we discussed, let's go with optional `--app`, `--cloud` etc when the time comes